node/src/chain/stall_chain_fsm.rs

```diff
@@ -205,29 +219,33 @@ impl<DB: database::DB, N: Network, VM: VMExecution> StalledChainFSM<DB, N, VM> {
         Ok(())
     }
 
-    /// Handles block acceptance timeout
-    ///
-    /// Request missing blocks since last finalized block
-    async fn on_accept_block_timeout(&mut self) -> Result<()> {
-        let (_, last_final_block) = self.last_final_block().await?;
-
-        let from = last_final_block + 1;
-        let to = self.tip.0.height + 1;
-
-        info!(event = "chain.requesting_blocks", from, to,);
-
-        let mut inv = Inv::new(0);
-        for height in from..to {
-            inv.add_block_from_height(height);
+    async fn on_heartbeat_in_stalled(&mut self) {
+        if let State::Stalled(timestamp) = self.state {
```

Comment:
We already know the state is Stalled here, since it's checked in `on_heartbeat_event`. Maybe consider removing the `if` condition

Reply:
Not a blocker (will be addressed in next PR)